ethdb/walk.go

```diff
@@ -93,39 +94,43 @@ func (sc *splitCursor) Next() (key1, key2, key3, val []byte, err error) {
 
 var EndSuffix = []byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff}
 
-func GetModifiedAccounts(db Getter, startTimestamp, endTimestamp uint64) ([]common.Address, error) {
-	keys := make(map[common.Address]struct{})
-	startCode := dbutils.EncodeTimestamp(startTimestamp)
-	if err := db.Walk(dbutils.PlainAccountChangeSetBucket, startCode, 0, func(k, v []byte) (bool, error) {
-		keyTimestamp, _ := dbutils.DecodeTimestamp(k)
+// GetModifiedAccounts returns a list of addresses that were modified in the block range
+func GetModifiedAccounts(tx Tx, startNum, endNum uint64) ([]common.Address, error) {
 
-		if keyTimestamp > endTimestamp {
-			return false, nil
+	changedAddrs := make(map[common.Address]struct{})
+	startCode := dbutils.EncodeTimestamp(startNum)
+
+	c := tx.Cursor(dbutils.PlainAccountChangeSetBucket)
```

Comment:
Add `defer c.Close()` after this to make sure cursor is closed when we get out of this function (for any reason)

Reply:
Learning moment...

Question for you. Is this just 'good practice' (in other words, the garbage collector collects it anyway) or will it leak memory or resources like 'C'?

Also, if I forget something like this is there some way to detect it later (other than search)?

Reply:
Garbage collector will collect "Golang" objects, but not the "C" objects attached to them. So most objects that are pure Go do not even have `Close` function, but if they do, that usually means you have to call it to free something. In our case, cursors may hold "C" objects (LMDB cursor if we use `--chaindata` in RPC daemon), and therefore have `Close()`